src/scripts/main.js

```diff
@@ -1,3 +1,74 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowButtons = document.querySelectorAll('[class*="row"]');
+const columnButtons = document.querySelectorAll('[class*="column"]');
+const MIN_PARTS = 2;
+const MAX_PARTS = 10;
+
+document.body.addEventListener('click', (e) => {
+  const button = e.target.closest('.button');
+
+  if (!button) {
+    return null;
+  }
+
+  const rows = [...table.querySelectorAll('tr')];
+  const column = rows[0].cells;
+  let rowsNumber = rows.length;
+  let columnNumber = column.length;
+  const rowProveRegex = /row/g;
+  const appendRegex = /append/g;
+  const buttonFunc = [...button.classList];
+  const containRows = buttonFunc.some((cls) => rowProveRegex.test(cls));
```

Comment:
Using regex for checking class names can be powerful and flexible, but it is often more complex and less readable than other approaches. Regex can be slightly slower than direct class name checks. For example we can check if it is rowOperation with `button.classList.contains('append-row') || button.classList.contains('remove-row');`